src/llmo-customer-analysis/utils.js

```diff
@@ -120,37 +139,94 @@ export function areCategoryNamesDifferent(oldCategories, newCategories) {
   const newNames = Object.values(newCategories || {}).map((c) => c?.name).filter(Boolean).sort();
   return oldNames.length !== newNames.length || !oldNames.every((name, i) => name === newNames[i]);
 }
+/**
+ * Checks if all changes in categories and topics are AI-origin only
+ */
+function checkAICategorizationOnly(changes, oldConfig, newConfig) {
+  if (!changes.categories && !changes.topics) return false;
+
+  // If there are other types of changes, not AI-only
+  const changeKeys = Object.keys(changes);
+  const hasOtherChanges = changeKeys.some((key) => key !== 'categories' && key !== 'topics');
+  if (hasOtherChanges) return false;
+
+  const oldTopics = oldConfig?.topics || {};
+  const newTopics = newConfig?.topics || {};
+  const oldAiTopics = oldConfig?.ai_topics || {};
+  const newAiTopics = newConfig?.ai_topics || {};
+
+  // Check new categories have topics with AI prompts
+  if (changes.categories) {
+    const newCategoryIds = Object.keys(changes.categories);
+    const topicsForNewCategories = [
+      ...Object.values(newTopics).filter((t) => newCategoryIds.includes(t.category)),
+      ...Object.values(newAiTopics).filter((t) => newCategoryIds.includes(t.category)),
+    ];
+
+    if (topicsForNewCategories.length === 0) return false;
+
+    for (const topic of topicsForNewCategories) {
+      const prompts = topic.prompts || [];
+      if (prompts.length === 0) return false;
+      if (prompts.some((p) => p.origin?.toLowerCase() !== 'ai')) return false;
+    }
+  }
+
+  // Check changed topics have only AI prompts
+  if (changes.topics) {
```

Comment:
don't ai prompts all go into `.aiPrompts`, and changed to `.prompts` mean you can `return false` immediately?